js/classes/entities/ui_ground/save-menu/login_screen.js

```diff
@@ -0,0 +1,244 @@
+        CTX.strokeRect(
+            this.start.x + TextInputField.LABEL_SIZE.x,
+            this.start.y,
+            TextInputField.FIELD_SIZE.x,
+            this.size.y
+        );
+        // draw the value in the box.
+        const text = this.pw ? '*'.repeat(this.value.length) : this.value;
+        CTX.fillText(
+            text,
+            this.start.x + TextInputField.LABEL_SIZE.x + 20,
+            this.start.y + 55,
+            TextInputField.FIELD_SIZE.x
+        );
+        this.trash.draw();
+    }
+}
+
+class LoginButton {
+    constructor(sy, parent) {
+        this.size = new Vector((PauseMenu.WIDTH * 2) / 5, 75);
+        this.pos = new Vector(
+            PauseMenu.START.x + (PauseMenu.WIDTH - this.size.x) / 2,
+            PauseMenu.START.y + sy
+        );
+        this.parent = parent;
+        const listener = () => {
+            const mouseOverTab =
+                GAME.mousePos.x > this.pos.x &&
+                GAME.mousePos.y > this.pos.y &&
+                GAME.mousePos.x < this.pos.x + this.size.x &&
+                GAME.mousePos.y < this.pos.y + this.size.y;
+            if (mouseOverTab) {
+                parent.attemptLogin();
+            }
+        };
+        document.body.addEventListener('click', listener);
+        HUD.componentListeners.push(['click', listener]);
+    }
+
+    draw() {
+        CTX.fillStyle = 'rgba(255,255,255,0.4)';
+        CTX.fillRect(this.pos.x, this.pos.y, this.size.x, this.size.y);
+        CTX.strokeStyle = 'rgb(255,255,255)';
+        CTX.lineWidth = 5;
+        CTX.strokeRect(this.pos.x, this.pos.y, this.size.x, this.size.y);
+        CTX.fillStyle = 'rgb(255,255,255)';
+        CTX.font = FONT.VT323_HEADER;
+        const login = 'LOG IN';
+        const textWidth = CTX.measureText(login).width;
+        CTX.fillText(
+            'LOG IN',
+            this.pos.x + (this.size.x - textWidth) / 2,
+            this.pos.y + 55,
+            this.size.x
+        );
+    }
+}
+
+class Feedback {
+    constructor(sy) {
+        this.fb = '';
+        this.sy = PauseMenu.START.y + sy;
+        this.mood = 0;
+    }
+
+    setFb(fb, mood = 0) {
+        this.fb = fb;
+        this.mood = mood;
+    }
+
+    draw() {
+        if (!this.fb) return;
+        CTX.font = FONT.VT323_NORMAL;
+
+        if (this.mood < 0) CTX.fillStyle = 'rgb(150, 0, 0)';
+        else if (this.mood > 0) CTX.fillStyle = 'rgb(0,150,0)';
+        else CTX.fillStyle = 'rgb(255,255,255)';
+
+        const width = CTX.measureText(this.fb).width;
+        CTX.fillText(
+            this.fb,
+            PauseMenu.START.x + (PauseMenu.WIDTH - width) / 2,
+            this.sy
+        );
+    }
+}
+
+class TrashCan {
+    constructor(sy, onClick) {
+        this.pos = new Vector(PauseMenu.START.x + PauseMenu.WIDTH - 100, sy);
+        this.size = new Vector(75, 75);
+        const listener = () => {
+            const mouseOver =
+                GAME.mousePos.x > this.pos.x &&
```

Comment:
I've seen similar code several times now, I wonder if it might be useful to make this large conditional into a generic function and reuse the function call instead.